### PR TITLE
fix: wond 5

### PR DESCRIPTION
**Pull request by 0xShaito**
# 🤖 Linear

Closes GRT-4

### CODE REVIEW

File: solidity/contracts/extensions/BondEscalationAccounting.sol [hunk 1]
```diff
@@ -24,13 +24,19 @@ contract BondEscalationAccounting is AccountingExtension, IBondEscalationAccount
...
     IERC20 _token,
     uint256 _amount
-  ) external onlyAllowedModule(_requestId) {
+  ) external {
     if (balanceOf[_pledger][_token] < _amount) revert BondEscalationAccounting_InsufficientFunds();
 
+    bytes32 _requestId = _getId(_request);
+    bytes32 _disputeId = _validateDispute(_request, _dispute);
+
+    if (ORACLE.disputeCreatedAt(_disputeId) == 0) revert BondEscalationAccounting_InvalidDispute();
```
Comment (0xng) [inline solidity/contracts/extensions/BondEscalationAccounting.sol:37]: i believe this is already checked on the `_validateDispute` function

Comment (0xShaito) [reply solidity/contracts/extensions/BondEscalationAccounting.sol:37]: Yes! removed!

**Review by 0xng — APPROVED** [review]

File: solidity/contracts/modules/resolution/PrivateERC20ResolutionModule.sol [hunk 2]
```diff
@@ -53,7 +53,6 @@ contract PrivateERC20ResolutionModule is Module, IPrivateERC20ResolutionModule {
   /// @inheritdoc IPrivateERC20ResolutionModule
   function commitVote(IOracle.Request calldata _request, IOracle.Dispute calldata _dispute, bytes32 _commitment) public {
     bytes32 _disputeId = _validateDispute(_request, _dispute);
-    if (ORACLE.disputeCreatedAt(_disputeId) == 0) revert PrivateERC20ResolutionModule_NonExistentDispute();
```
Comment (ashitakah) [inline solidity/contracts/modules/resolution/PrivateERC20ResolutionModule.sol:56]: Why this check is deleted?

File: solidity/test/unit/modules/dispute/BondEscalationAccounting.t.sol [hunk 3]
```diff
@@ -107,41 +107,52 @@ contract BaseTest is Test, Helpers {
...
 
 contract BondEscalationAccounting_Unit_Pledge is BaseTest {
-  function test_revertIfDisallowedModule(
-    address _pledger,
-    bytes32 _requestId,
-    bytes32 _disputeId,
-    uint256 _amount
-  ) public {
+  function test_revertIfDisallowedModule(address _pledger, uint256 _amount) public {
+    (IOracle.Response memory _response, IOracle.Dispute memory _dispute) = _getRequestAndDispute();
```
Comment (ashitakah) [inline solidity/test/unit/modules/dispute/BondEscalationAccounting.t.sol:128]: linter

**Review by ashitakah — COMMENTED** [review]
Just a few comments, looks solved it to me.

File: solidity/contracts/modules/resolution/PrivateERC20ResolutionModule.sol (hunk 2 above)
Comment (0xShaito) [reply solidity/contracts/modules/resolution/PrivateERC20ResolutionModule.sol:56]: Because it's done in the validateDispute

Comment (0xJabberwock) [reply solidity/contracts/modules/resolution/PrivateERC20ResolutionModule.sol:56]: In consequence, `PrivateERC20ResolutionModule_NonExistentDispute` error could be removed from `IPrivateERC20ResolutionModule`.

**Review by 0xJabberwock — CHANGES_REQUESTED** [review]

File: solidity/contracts/extensions/AccountingExtension.sol [hunk 4]
```diff
@@ -6,15 +6,14 @@ import {IERC20} from '@openzeppelin/contracts/token/ERC20/IERC20.sol';
 import {SafeERC20} from '@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol';
 import {EnumerableSet} from '@openzeppelin/contracts/utils/structs/EnumerableSet.sol';
 
+import {Validator} from '@defi-wonderland/prophet-core-contracts/solidity/contracts/Validator.sol';
+
 import {IAccountingExtension} from '../../interfaces/extensions/IAccountingExtension.sol';
 
-contract AccountingExtension is IAccountingExtension {
+contract AccountingExtension is Validator, IAccountingExtension {
```
Comment (gas1cent) [inline solidity/contracts/extensions/AccountingExtension.sol:13]: Such a great idea to extract the validation logic into a contract.

File: solidity/contracts/extensions/AccountingExtension.sol
[the remark this answers is not in the document]
Comment (0xShaito) [reply solidity/contracts/extensions/AccountingExtension.sol:1]: This particular exploit does not affect the accounting extension as it does not depend on disputes

**Review by 0xJabberwock — APPROVED** [review]

**Review by ashitakah — APPROVED** [review]

**Pull request merged**